### PR TITLE
Update annual-report-2023.md

### LINKED ISSUE #7762: 2023 Annual Report: SIG API Machinery
2023 Annual Report: SIG API Machinery

Chairs: @deads2k @fedebongio 
Liaison: @palnabarun

Actions for the chair/organizer of the community group:
- [ ] Consult your community group to complete draft of report
- [ ] If needed, consult with Steering Committee liaison on any private issues or concerns that arrise while completing report
- [ ] Submit PR to git.k8s.io/community repo with copy of report, assigning your Steering Committee liaison for review
- [ ] Steering Committee reviews report, and may make comments/ask questions
- [ ] Once all comments are addressed, Steering Committee will approve report to merge
- [ ] If needed, any follow up items may be brought to the "Chairs, Tech Leads, and Organizers" meeting in April

Once all the above items are complete, this issue may be `/close`'d

Key dates:
- Initial PR to communtiy repo should be opened by May 1, 2024
- PR should be reviewed and merged by May 22, 2024

More detailed information on the annual reports process is available [here](https://git.k8s.io/community/committee-steering/governance/annual-reports.md).

Your group's report template is available here: https://git.k8s.io/community/sig-api-machinery/annual-report-2023.md

If you have any questions or concerns about this process, you may reach the Steering Committee via the following methods:
- Slack/E-mail your liaison directly
- The public #steering-committee channel in Slack
- The public mailing list steering@kubernetes.io
- The private mailing list steering-private@kubernetes.io for any private or sensitive issues.

/assign @deads2k @fedebongio 
/sig api-machinery
/committee steering
/area annual-reports

### PR DESCRIPTION
**Pull request by fedebongio**
**Which issue(s) this PR fixes**:

Fixes #7762

### CODE REVIEW

**Comment by sttts** [comment]
Lgtm

File: sig-api-machinery/annual-report-2023.md [hunk 1]
```diff
@@ -14,17 +14,22 @@
```
Comment (palnabarun) [inline sig-api-machinery/annual-report-2023.md:14]: @kubernetes/sig-api-machinery-leads this section needs to be populated.

File: sig-api-machinery/annual-report-2023.md [hunk 2]
```diff
@@ -14,17 +14,22 @@
 
 2. Are there any areas and/or subprojects that your group needs help with (e.g. fewer than 2 active OWNERS)?
 
-<!--
-   Note: This list is generated from the KEP metadata in kubernetes/enhancements repository.
-      If you find any discrepancy in the generated list here, please check the KEP metadata.
-      Please raise an issue in kubernetes/community, if the KEP metadata is correct but the generated list is incorrect.
--->
+Yes, in general SIG API Machinery owns a large portion of the Kubernetes codebase. We are always looking to welcome more contributors that help up improve and keep healthy this core important part of Kubernetes.
+Just to mention a few areas:
```
Comment (palnabarun) [inline sig-api-machinery/annual-report-2023.md:18]: Could you also please link to some issues/documents that can help a prospective contributor to get started?

File: sig-api-machinery/annual-report-2023.md [hunk 3]
```diff
@@ -14,17 +14,22 @@
...
+- server side apply
+- documentation
+- and many others
 
 3. Did you have community-wide updates in 2023 (e.g. KubeCon talks)?
 
-<!--
-  Examples include links to email, slides, or recordings.
--->
+Yes, there was a deep dive [API Machinery session](https://kccncna2023.sched.com/event/1R2my) at Kubecon NA 2023 that covered:
```
Comment (palnabarun) [inline sig-api-machinery/annual-report-2023.md:27]: Was there a session at KubeCon EU 2023?

File: sig-api-machinery/annual-report-2023.md [hunk 4]
```diff
@@ -48,24 +53,10 @@
...
-  - idl-schema-client-pipeline
-  - json
-  - kubernetes-clients
-  - server-api-aggregation
-  - server-binaries
-  - server-crd
-  - server-frameworks
-  - server-sdk
-  - universal-machinery
-  - yaml
```
Comment (palnabarun) [inline sig-api-machinery/annual-report-2023.md:68]: This is generated. Please revert the removal of lines.

File: sig-api-machinery/annual-report-2023.md [hunk 5]
```diff
@@ -48,24 +53,10 @@
...
+Highlights of the work from this list:
+- The work done in Priority and Fairnes that led to Stable
+- The work done on ValidatingAdmissionControl based on CEL, that will later on graduate to Stable in 1.30
```
Comment (palnabarun) [inline sig-api-machinery/annual-report-2023.md:58]: This section should probably be bubbled up to Answer for Question 1.

File: sig-api-machinery/annual-report-2023.md [hunk 6]
```diff
@@ -14,17 +14,22 @@
...
-  Examples include links to email, slides, or recordings.
--->
+Yes, there was a deep dive [API Machinery session](https://kccncna2023.sched.com/event/1R2my) at Kubecon NA 2023 that covered: 
+- An introduction to SIG API Machinery
+- An overview of Generic Control Planes
+- A deep Dive into CRD Ratcheting
+
+We also recorded a spotlight for Kubecon NA that was projected before keynotes.
 
 4. KEP work in 2023 (v1.27, v1.28, v1.29):
```
Comment (palnabarun) [inline sig-api-machinery/annual-report-2023.md:34]: This list of KEPs needs to be reconfirmed.

File: sig-api-machinery/annual-report-2023.md [hunk 7]
```diff
@@ -48,24 +53,10 @@
     - [2896 - OpenAPI V3](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/2896-openapi-v3) - v1.27
     - [365 - Paginated API Lists](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/365-paginated-lists) - v1.29
 
-## [Subprojects](https://git.k8s.io/community/sig-api-machinery#subprojects)
```
Comment (jpbetz) [inline sig-api-machinery/annual-report-2023.md:51]: I reviewed the KEP list against my personal notes.

- Generic Control Plane started - https://github.com/kubernetes/enhancements/pull/4052 - alpha 1.27 (I believe)
- Move Storage Version Migrator in-tree - https://github.com/kubernetes/enhancements/pull/4240 - alpha 1.29

File: sig-api-machinery/annual-report-2023.md (hunk 3 above)
Comment (sttts) [reply sig-api-machinery/annual-report-2023.md:27]: There wasn't.

**Pull request closed without merging**